data/basics/overflow.go

```diff
@@ -173,28 +130,28 @@ func MulAIntSaturate(a MicroAlgos, b int) MicroAlgos {
 }
 
 // AddA adds 2 MicroAlgos values with overflow tracking
-func (t *OverflowTracker) AddA(a MicroAlgos, b MicroAlgos) MicroAlgos {
-	return MicroAlgos{Raw: t.Add(uint64(a.Raw), uint64(b.Raw))}
+func (t *OverflowTracker[T]) AddA(a, b MicroAlgos) MicroAlgos {
+	return MicroAlgos{Raw: uint64(t.Add(T(a.Raw), T(b.Raw)))}
 }
 
 // SubA subtracts b from a with overflow tracking
-func (t *OverflowTracker) SubA(a MicroAlgos, b MicroAlgos) MicroAlgos {
-	return MicroAlgos{Raw: t.Sub(uint64(a.Raw), uint64(b.Raw))}
+func (t *OverflowTracker[T]) SubA(a, b MicroAlgos) MicroAlgos {
+	return MicroAlgos{Raw: uint64(t.Sub(T(a.Raw), T(b.Raw)))}
 }
 
 // AddR adds 2 Round values with overflow tracking
-func (t *OverflowTracker) AddR(a Round, b Round) Round {
-	return Round(t.Add(uint64(a), uint64(b)))
+func (t *OverflowTracker[T]) AddR(a, b Round) Round {
```

Comment:
Can you look around a bit and see if you see places where there should have been used, but maybe people didn't know about them, so cast back and forth to uint64?

If not, I agree with deletion.

(In fact, I think that generics can take AddR's place - see if you can Add() rounds with your change. I believe that `constraints.Unsigned` works on anything with a unsigned int - like underlying type, which Round has.)

Reply:
right, I am leaning towards using `OAdd` to replace `AddR`, for one can use the generics to compute over `Round`, since `Round` itself is `constraints.Unsigned`.

I deleted the `AddR` impl in [remove round addr subr for it can be done through oadd or osub](https://github.com/algorand/go-algorand/pull/5501/commits/67d0fbe87a17e61aff6f93449403d1aa0dbaf500)